raiden/transfer/token_network.py

```diff
@@ -254,30 +255,19 @@ def handle_batch_unlock(
     block_number: BlockNumber,
     block_hash: BlockHash,
 ) -> TransitionResult:
-    events = list()
-    channel_state = None
-    if token_network_state.channelidentifiers_to_channels.get(state_change.participant) is not None:
-        channel_state = token_network_state.channelidentifiers_to_channels[state_change.participant].get(
-            state_change.canonical_identifier.channel_identifier
-        )
-    if channel_state is not None:
-        sub_iteration = channel.state_transition(
-            channel_state=channel_state,
-            state_change=state_change,
-            block_number=block_number,
-            block_hash=block_hash,
-        )
-        events.extend(sub_iteration.events)
-
-        if sub_iteration.new_state is None:
-
-            token_network_state.partneraddresses_to_channelidentifiers[
-                channel_state.partner_state.address
-            ].remove(channel_state.identifier)
-
-            del token_network_state.channelidentifiers_to_channels[state_change.participant][channel_state.identifier]
+    participant = None
+    if state_change.participant in token_network_state.channelidentifiers_to_channels:
+        participant = state_change.participant
+    elif state_change.partner in token_network_state.channelidentifiers_to_channels:
+        participant = state_change.partner
```

Comment:
are you aware that if both of these equal `False`, participant will be `None`?

this will apparently not cause an exception in `subdispatch_to_channels_by_participant_address`.

are we OK with this? (honestly don't know)

Reply:
yeah, this is important, because before this change it was not calling the state transition function for None channels

Reply:
when the participant address is None, the subdispatch_to_channels_by_participant_address does nothing. Maybe we should remove the optionallity of that param someday